pkg/overlay/overlay.go

```diff
@@ -55,6 +48,36 @@ func NewOverlayReader(base io.ReadSeeker, overlay Overlay) (io.ReadSeeker, error
 	return &or, nil
 }
 
+func NewOverlayReader(base io.ReadSeeker, overlay Overlay) (io.ReadSeeker, error) {
+	length, err := base.Seek(0, io.SeekEnd)
+	if err != nil {
+		return nil, err
+	}
+	if overlay.Offset < 0 || overlay.Offset > length {
+		return nil, errors.New("Overlay offset is beyond end of base")
+	}
+	return newReader(base, overlay, length)
+}
+
+func NewAppendReader(base io.ReadSeeker, reader io.ReadSeeker) (io.ReadSeeker, error) {
```

Comment:
Can't we just use a [MultiReader](https://pkg.go.dev/io@go1.17.3#MultiReader) for this?

Reply:
I was hoping so too, but it returns a `Reader` interface instead of `ReadSeeker` because golang :roll_eyes:

Reply:
:man_facepalming: right.